code/modules/mining/minebot.dm

```diff
@@ -1,313 +1,699 @@
-	Field repairs can be done with a welder."}
-	if(stored_gun && stored_gun.max_mod_capacity)
-		. += "<b>[stored_gun.get_remaining_mod_capacity()]%</b> mod capacity remaining."
-		for(var/A in stored_gun.get_modkits())
+			. += "<span class='boldwarning'>It is disabled and requires repairs.</span>"
+	. += "<span class='notice'>Alt-clicking it will instruct it to drop any stored ore.</span>"
+	. += "<span class='notice'>Field repairs can be performed with a welder.</span>"
+	if(stored_pka && stored_pka.max_mod_capacity)
+		. += "<span class='notice'>\The [stored_pka] has <b>[stored_pka.get_remaining_mod_capacity()]%</b> mod capacity remaining.</span>"
+		for(var/A as anything in stored_pka.get_modkits())
 			var/obj/item/borg/upgrade/modkit/M = A
 			. += "<span class='notice'>There is \a [M] installed, using <b>[M.cost]%</b> capacity.</span>"
-
-/mob/living/simple_animal/hostile/mining_drone/welder_act(mob/living/user, obj/item/I)
-	. = TRUE
-	if(mode == MINEDRONE_ATTACK)
-		to_chat(user, "<span class='info'>[src] can't be repaired while in attack mode!</span>")
+	if(stored_cutter)
+		. += "<span class='notice'>There is \a [stored_cutter] installed on its plasma cutter mount. The charge meter reads [round(stored_cutter.cell.percent())]%.</span>"
+	else
+		. += "<span class='notice'>There is nothing on its plasma cutter mount.</span>"
+	. += "<span class='notice'>There is \a [stored_drill] installed on its drill mount.</span>"
+	if(client)
+		. += "<span class='notice'>Its AI light is on.</span>"
+
+/// Generates the stat tab for player-controlled minebots
+/mob/living/simple_animal/hostile/mining_drone/get_stat_tab_status()
+	var/list/tab_data = ..()
+	tab_data["Mode"] = GENERATE_STAT_TEXT("[mode]")
+
+	// Handles Equipment
+	if(stored_cutter)
+		tab_data["Plasma Cutter Charge"] = GENERATE_STAT_TEXT("[round(stored_cutter.cell.percent())]%")
+	tab_data["Equipped Drill"] = GENERATE_STAT_TEXT("[stored_drill]")
+
+	// Handles Upgrades
+	if(LAZYLEN(installed_upgrades))
+		for(var/obj/item/minebot_upgrade/U as anything in installed_upgrades)
+			var/upgrade_data = U.get_stat_data()
+			if(upgrade_data)
+				tab_data[upgrade_data[1]] = GENERATE_STAT_TEXT(upgrade_data[2])
+	return tab_data
+
+/// Repairing/reviving minebots
+/mob/living/simple_animal/hostile/mining_drone/welder_act(mob/living/user, obj/item/welder)
+	if(istype(welder, /obj/item/gun/energy/plasmacutter) && maxHealth == health) // So we don't show the welding message while installing a plasma cutter
 		return
-
 	if(maxHealth == health)
 		to_chat(user, "<span class='info'>[src] is at full integrity.</span>")
 		return
-
-	if(I.use_tool(src, user, 0, volume=40))
+	if(welder.use_tool(src, user, 0, volume = 40))
+		if(stat == DEAD)
+			adjustBruteLoss(-25)
+			to_chat(user, "<span class='info'>You repair and restart [src].</span>")
+			revive()
```

Comment:
This should only revive if they have enough health to not instantly die

Reply:
Changed it to function that way.